src/main/java/uk/gov/ons/ctp/response/action/service/impl/decorator/PartyActionRequestDecorator.java

```diff
@@ -0,0 +1,157 @@
+package uk.gov.ons.ctp.response.action.service.impl.decorator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionContact;
+import uk.gov.ons.ctp.response.action.message.instruction.ActionRequest;
+import uk.gov.ons.ctp.response.action.service.ActionProcessingService;
+import uk.gov.ons.ctp.response.action.service.ActionRequestDecorator;
+import uk.gov.ons.ctp.response.party.representation.Association;
+import uk.gov.ons.ctp.response.party.representation.Attributes;
+import uk.gov.ons.ctp.response.party.representation.Enrolment;
+import uk.gov.ons.ctp.response.party.representation.PartyDTO;
+
+@Slf4j
+public class PartyActionRequestDecorator implements ActionRequestDecorator {
+
+  @Override
+  public void decorateActionRequest(
+      ActionRequest actionRequest, ActionRequestDecoratorContext context) {
+    final PartyDTO parentParty = context.getParentParty();
+    final List<PartyDTO> childParties = context.getChildParties();
+    final Attributes businessUnitAttributes = parentParty.getAttributes();
+    actionRequest.setRegion(businessUnitAttributes.getRegion());
+
+    final ActionContact actionContact = new ActionContact();
+    actionContact.setRuName(businessUnitAttributes.getName());
+    actionContact.setTradingStyle(generateTradingStyle(businessUnitAttributes));
+
+    if (context.getSampleUnitType().isParent() == false) {
```

Comment:
```
boolean isIndividual = context.getSampleUnitType().isParent()
if (isIndividual) {
```
would make this check more self-explanatory

Reply:
Done